nuxeo-core/nuxeo-core-cache/src/main/java/org/nuxeo/ecm/core/cache/CacheServiceImpl.java

```diff
@@ -189,7 +189,7 @@ public int getApplicationStartedOrder() {
     public void start(ComponentContext context) {
         super.start(context);
         ClusterService clusterService = Framework.getService(ClusterService.class);
-        if (clusterService.isEnabled()) {
+        if (clusterService != null && clusterService.isEnabled()) {
```

Comment:
Please use NXP-25499 for this fix.
But the fix should rather be to add the proper `@Deploy("org.nuxeo.runtime.cluster")` to the test that fails (until we have cleaned up all our features to have one feature per test service).

Reply:
Ok for the NXP-25499
For the fix, I do not agree, it's a potential NPE if clusterService is not deployed tests or not, and apparently it's not a blocking case

Reply:
ClusterService is a basic runtime service that is required by pubsub and therefore for invalidations of caches, directories and repositories. It should always be deployed before the cache service.

Reply:
If we had proper dependencies between services we would make this explicit with injection or requirements...